atom/browser/net/url_request_async_asar_job.cc

```diff
@@ -38,6 +38,22 @@ void BeforeStartInUI(base::WeakPtr<URLRequestAsyncAsarJob> job,
     error = net::ERR_NOT_IMPLEMENTED;
   }
 
+  // sanitize custom headers
+  if (request_options && request_options->is_dict()) {
+    base::DictionaryValue* dict =
+        static_cast<base::DictionaryValue*>(request_options.get());
```

Comment:
We shouldn't have to static_cast the `base::Value` to a `DictionaryValue` here.  Once you've checked `is_dict` you should be able to use the `Value` methods.

Refs: https://cs.chromium.org/chromium/src/base/values.h?type=cs&q=setboolkey&sq=package:chromium&g=0&l=272-325

Reply:
Same thing happens in `StartAsync`- should i simplify the code there too?